### ISSUE TITLE
Race condition in creating & updating users/roles

### ISSUE THREAD

**Issue by phillbaker** [opendistro-for-elasticsearch/security#1077]
Hello, I help maintain a terraform provider for elasticsearch, including OpenDistro security resources like users and roles.

We've received a couple of reports that seems to be a race condition in OpenDistro, here's [an example](https://github.com/phillbaker/terraform-provider-elasticsearch/issues/154). I was able to [reproduce this consistently in a CI test](https://github.com/phillbaker/terraform-provider-elasticsearch/commit/b397f19678cc24cd6625b7872b181106ceb1cb6f).

To reproduce this:
* have a script create 3 users in parallel
* one of the users will be created, the others will error with responses like:
```
{"status":"INTERNAL_SERVER_ERROR","message":"Error [internalusers]: version conflict, required seqNo [31], primary term [1]. current document has seqNo [32] and primary term [1]"}
```

This seems to be a race condition, adding a function to slow down requests on the provider side works around the error response. My guess is updates to the internalusers needs to be synchronized to make the updates sequential and avoid having external parallel requests fail.

**Comment by vrozov** [comment]
This is an expected behavior. It is not possible to synchronize in the distributed environment and plugin uses optimistic concurrency control to update users, roles and other objects. Please add error handling to your scripts.

**Comment by phillbaker** [comment]
Thanks for taking a look @vrozov - in the case that it is an expected failure, can we re-open this or create a new issue for the API to return a HTTP 409 (conflict) status code? That would align the behavior here with other elasticsearch APIs.

**Comment by vrozov** [comment]
@phillbaker Please open a new issue or re-open this one to look into the status code.
